operator/internal/manifests/gateway.go

```diff
@@ -58,6 +58,9 @@ func BuildGateway(opts Options) ([]client.Object, error) {
 		}
 
 		objs = configureGatewayObjsForMode(objs, opts)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
I believe this err check is obsolete now as per `configureGatewayObjsForMode` not returning an error value anymore.

Reply:
nice catch!